python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -0,0 +1,124 @@
+from typing import Callable, Optional, Union, List, Type
+
+import tensorflow as tf
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+# TensorFlow model objects cannot be pickled, therefore we use
+# a callable that returns the model, instead of a model object
+# itself.
+
+
+class TensorflowPredictor(Predictor):
+    """A predictor for TensorFlow models.
+
+    Args:
+        model_definition: A callable that returns a TensorFlow Keras model
+            to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+        model_weights: List of weights to use for the model.
+    """
+
+    def __init__(
+        self,
+        model_definition: Union[Callable[[], tf.keras.Model], Type[tf.keras.Model]],
```

Comment:
Q: how does this relate to SavedModel format?

Reply:
No support for it yet, though we could add it in the future